clean_air/util/file_converter.py

```diff
@@ -0,0 +1,171 @@
+    # use as a dataframe.
+    form_responses = list()
+    for r, row in enumerate(dataframe.iterrows()):
+        # iterate through each row to split into separate dataframes and save
+        # all to appropriate location:
+        form_data = {'title': row[1].values[16],
+                     'description': row[1].values[17],
+                     'firstname1': row[1].values[6],
+                     'surname1': row[1].values[7],
+                     'firstname2': row[1].values[11],
+                     'surname2': row[1].values[12],
+                     'north': row[1].values[42],
+                     'south': row[1].values[41],
+                     'east': row[1].values[43],
+                     'west': row[1].values[44],
+                     'chemicals': row[1].values[46].split(';'),
+                     'obs_level': row[1].values[19],
+                     'data_source': row[1].values[20],
+                     'time_range_start': row[1].values[37],
+                     'time_range_end': row[1].values[38],
+                     'lineage': row[1].values[50],
+                     'quality': row[1].values[51],
+                     'docs': row[1].values[22]}
+
+        form = pd.Series(data=form_data)
+        form_responses.append([form, r])
+
+    return form_responses
+
+
+def save_as_json(data_object, r, output_location):
+    """
+    Uses data held in pandas DataFrame to enter into form template and
+    save as JSON string.
+    """
+    # Pollutants(chemicals) must be organised into a useable structure before
+    # being entered into the new file:
+    chemicals = []
+    for chem in data_object.chemicals:
+        if chem != "":
+            chems = {'name': chem,
+                     'shortname': chem[chem.find("(") + 1:chem.find(")")],
+                     'chart': "url/to/chart/image.(png|jpg|svg|etc)"}
+            chemicals.append(chems)
+
+    new_file = {"pollutants": chemicals,
+                "environmentType": data_object.obs_level,
+                "dateRange": {"startDate": data_object.time_range_start,
+                              "endDate": data_object.time_range_end}}
+
+    # write the dictionary above (new_file) to a json with the addition of
+    # a unit to indicate the number of the metadata form response.
+    filename = "form_response" + str(r) + ".json"
+    with open(os.path.join(output_location, filename), 'w') as fp:
+        json.dump(new_file, fp, indent=2, cls=DateTimeEncoder)
+
+
+def save_as_yaml(data_object, r, output_location):
+    """
+    Uses data held in pandas DataFrame to enter into form template and
+    save as yaml.  Each data object entered here should be a single
+    response from the previous form input section.
+    """
+    # First extract the shortname only for chemical species:
+    chem_species = []
+    for chem in data_object.chemicals:
+        if chem != "":
+            chem_shortname = chem[chem.find("(") + 1:chem.find(")")]
```

Comment:
I think it's actually still needed to find the shortened version of the chemical name, so it will find the bits inside the brackets (i.e. "PM10" as opposed to "Coarse Particulate Matter (PM10)").